### PR TITLE
dumpling: update dumpling document on how to reduce memory usage

### PR DESCRIPTION
**Pull request by lichunzhu**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Update dumpling document on how to reduce memory usage. Make sure users can get this important infomation.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@lichunzhu Please involve a technical review~

File: dumpling-overview.md [hunk 1]
```diff
@@ -60,10 +60,18 @@ dumpling \
   --filetype sql \
   --threads 32 \
   -o /tmp/test \
+  -r 200000 \
```
Comment (overvenus) [inline dumpling-overview.md:63]: I think it's too large and may cause TiDB OOM.

Comment (overvenus) [inline dumpling-overview.md:63]: Please also add `-r` to other examples, e.g., `### 导出到 csv 文件`

Comment (lichunzhu) [reply dumpling-overview.md:63]: If we suggest TiDB uses 1GB memory, and the default threads is 4, then avg_row_length is 
1GB / 4 / 200000 = 1342. It's more than 1KB. In most conditions `avg_row_length` is less than this.

Comment (lichunzhu) [reply dumpling-overview.md:63]: added in https://github.com/pingcap/docs-cn/pull/5860/commits/0c0706ba99daad3dcbe3e8c18596791287ad6580

File: dumpling-overview.md [hunk 2]
```diff
@@ -60,10 +60,18 @@ dumpling \
...
 
-上述命令中，`-h`、`-P`、`-u` 分别是地址，端口，用户。如果需要密码验证，可以用 `-p $YOUR_SECRET_PASSWORD` 传给 Dumpling。
+- `-h`、`-P`、`-u` 分别是地址，端口，用户。如果需要密码验证，可以用 `-p $YOUR_SECRET_PASSWORD` 传给 Dumpling。
+- `-o` 用于选择存储导出文件的目录，支持本地文件路径或[外部存储 URL](/br/backup-and-restore-storages.md) 格式。
+- `-r` 可以指定单个文件的最大行数，开启后 Dumpling 会开启表内并发加速导出，同时减少内存使用。
+- `-F` 可以指定单个文件的最大大小。
+
+> **注意：**
+>
+> 导出较大表（单表大小超过 10 GB）时 `-r`, `-F` 参数均**强烈建议**开启。
```
Comment (3pointer) [inline dumpling-overview.md:74]: -r -F 不能同时生效，也描述下吧

Comment (lichunzhu) [reply dumpling-overview.md:74]: 现在是可以同时生效的

**Comment by overvenus** [comment]
@TomShawn PTAL, thanks!

File: dumpling-overview.md [hunk 3]
```diff
@@ -60,10 +60,18 @@ dumpling \
   --filetype sql \
   --threads 32 \
   -o /tmp/test \
+  -r 200000 \
   -F 256MiB
 ```
 
-上述命令中，`-h`、`-P`、`-u` 分别是地址，端口，用户。如果需要密码验证，可以用 `-p $YOUR_SECRET_PASSWORD` 传给 Dumpling。
+- `-h`、`-P`、`-u` 分别是地址，端口，用户。如果需要密码验证，可以用 `-p $YOUR_SECRET_PASSWORD` 传给 Dumpling。
```
Comment (TomShawn) [inline dumpling-overview.md:67]:
```suggestion
以上命令中：

- `-h`、`-P`、`-u` 分别代表地址、端口、用户。如果需要密码验证，可以使用 `-p $YOUR_SECRET_PASSWORD` 将密码传给 Dumpling。
```

File: dumpling-overview.md [hunk 4]
```diff
@@ -60,10 +60,18 @@ dumpling \
...
+- `-o` 用于选择存储导出文件的目录，支持本地文件路径或[外部存储 URL](/br/backup-and-restore-storages.md) 格式。
+- `-r` 可以指定单个文件的最大行数，开启后 Dumpling 会开启表内并发加速导出，同时减少内存使用。
```
Comment (TomShawn) [inline dumpling-overview.md:69]:
```suggestion
- `-r` 用于指定单个文件的最大行数，指定该参数后 Dumpling 会开启表内并发加速导出，同时减少内存使用。
```

File: dumpling-overview.md (hunk 2 above)
Comment (TomShawn) [inline dumpling-overview.md:74]:
```suggestion
> 如果导出的单表大小超过 10 GB，**强烈建议**使用`-r` 和 `-F` 参数。
```

Comment (TomShawn) [inline dumpling-overview.md:74]:
```suggestion
> 如果导出的单表大小超过 10 GB，**强烈建议**指定 `-r` 和 `-F` 参数。
```

File: dumpling-overview.md [hunk 5]
```diff
@@ -236,11 +246,10 @@ Dumpling 也可以通过 `-B` 或 `-T` 选项导出特定的数据库/数据表
 
 默认情况下，导出的文件会存储到 `./export-<current local time>` 目录下。常用选项如下：
 
-- `-o` 用于选择存储导出文件的目录。
-- `-F` 选项用于指定单个文件的最大大小，默认单位为 `MiB`。可以接受类似 `5GiB` 或 `8KB` 的输入。
+- `-t` 用于指定导出线程数，增大线程数会提高 Dumpling 并发度，但也会加大数据库内存消耗，因此不宜过大。
```
Comment (TomShawn) [inline dumpling-overview.md:249]:
```suggestion
- `-t` 用于指定导出的线程数。增加线程数会增加 Dumpling 并发度，但也会加大数据库内存消耗，因此不宜设置过大。
```

File: dumpling-overview.md [hunk 6]
```diff
@@ -236,11 +246,10 @@ Dumpling 也可以通过 `-B` 或 `-T` 选项导出特定的数据库/数据表
...
 - `-r` 选项用于指定单个文件的最大记录数（或者说，数据库中的行数），开启后 Dumpling 会开启表内并发，提高导出大表的速度。
 
-利用以上选项可以让 Dumpling 的并行度更高。
+利用以上选项可以提高 Dumpling 导出速度。
```
Comment (TomShawn) [inline dumpling-overview.md:252]:
```suggestion
利用以上选项可以提高 Dumpling 的导出速度。
```

**Pull request merged**